### PR TITLE
Detect if video files are stores on Django Storages' S3

### PR DESCRIPTION
**Pull request by Timusan**
Fixes the issue mentioned in #106.

Transcoding will now check if the file is stored using Django's Boto3 backend.
If so it will use its URL instead of its absolute path.

Tied and tested. Not sure if this scales to all possible causes of the `NotImplementedError` exception though.

### CODE REVIEW

File: wagtailvideos/models.py [hunk 1]
```diff
@@ -256,9 +256,23 @@ def __init__(self, transcode, **kwargs):
...
+        except ImportError:
+            S3Boto3Storage = False
+
         video = self.transcode.video
         media_format = self.transcode.media_format
-        input_file = video.file.path
+
+        # If the file is stored on S3, we need to use the URL instead of the path.
+        storage_backend = video.file.storage
+        if S3Boto3Storage and isinstance(storage_backend, S3Boto3Storage):
```
Comment (seb-b) [inline wagtailvideos/models.py:271]: We can probably condense this down a bit to remove the import check above -

```python
try:
    input_file = video.file.path
except NotImplementedError:
    input_file = vide.file.url 
```

I had no idea ffmpeg works with a url as well as a file path. I'm assuming this all works ok? I don't really have an easy way to test it

Comment (Timusan) [reply wagtailvideos/models.py:271]: Maybe, the condensed version is what I had in the beginning, but I am not sure that `NotImplementedError` is always due to the absolute path exception. In other words, I feel this is a bit too broad and does not guarantee we have `video.file.url` available. That is why I check if Storages is installed and specifically check on the S3 backend being set so I know for sure that we have a `url` attribute on `video.file`.

And yes, this is tested, ffmpeg accepts URLs just fine as input.

Comment (seb-b) [reply wagtailvideos/models.py:271]: I'm ok with the broad approach, it's basically the same check that Wagtail does - https://github.com/wagtail/wagtail/blob/main/wagtail/images/models.py#L133

**Comment by seb-b** [comment]
duplicate of #111

**Pull request closed without merging**